pkg/rules/remotewrite/wal.go

```diff
@@ -0,0 +1,702 @@
+// This is copied from https://github.com/grafana/agent/blob/a23bd5cf27c2ac99695b7449d38fb12444941a1c/pkg/prom/wal/wal.go
+// TODO(idoqo): Migrate to prometheus package when https://github.com/prometheus/prometheus/pull/8785 is ready.
+package remotewrite
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/pkg/value"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/prometheus/prometheus/tsdb/record"
+	"github.com/prometheus/prometheus/tsdb/wal"
+)
+
+// ErrWALClosed is an error returned when a WAL operation can't run because the
+// storage has already been closed.
+var ErrWALClosed = fmt.Errorf("WAL storage closed")
+
+type storageMetrics struct {
+	r prometheus.Registerer
+
+	numActiveSeries      prometheus.Gauge
+	numDeletedSeries     prometheus.Gauge
+	totalCreatedSeries   prometheus.Counter
+	totalRemovedSeries   prometheus.Counter
+	totalAppendedSamples prometheus.Counter
+}
+
+func newStorageMetrics(r prometheus.Registerer) *storageMetrics {
+	m := storageMetrics{r: r}
+	m.numActiveSeries = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "agent_wal_storage_active_series",
```

Comment:
```suggestion
		Name: "wal_storage_active_series",
```

Let's start with `wal...`? Let's remove `agent`